### PR TITLE
feat(DHT): Enforce node id length

### PR DESCRIPTION
**Pull request by juslesan**
## Summary

If an explicitly configured entry point is not a hex string of length 40 the DhtNode will throw

Bump `@streamr/config` to `^5.1.2`

### CODE REVIEW

File: packages/broker/configs/docker-3.env.json [hunk 1]
```diff
@@ -7,15 +7,15 @@
...
+                    "nodeId": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb3",
                     "websocket": {
                         "host": "10.200.10.1",
                         "port": 40403,
                         "tls": false
                     }
                 },
                 "entryPoints": [{
-                    "nodeId": "eeeeeeeeee",
+                    "nodeId": "eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
```
Comment (teogeb) [inline packages/broker/configs/docker-3.env.json]: Should this be `40` chars long? (i.e. same as in `development-3.json`)

File: packages/dht/src/dht/DhtNode.ts [hunk 2]
```diff
@@ -143,10 +143,21 @@ export class DhtNode extends EventEmitter<Events> implements ITransport {
...
+        this.validateConfig()
         this.localDataStore = new LocalDataStore(this.config.storeMaxTtl)
         this.send = this.send.bind(this)
     }
 
+    private validateConfig(): void {
+        if (this.config.nodeId !== undefined ) {
+            if (!/^[0-9a-fA-F]+$/.test(this.config.nodeId)) {
+                throw new Error('Invalid nodeId, the nodeId should be a hex string')
+            } else if (this.config.nodeId.length !== 40) {
```
Comment (teogeb) [inline packages/dht/src/dht/DhtNode.ts]: Could use K`ADEMLIA_ID_LENGTH_IN_BYTES*2`

**Review by teogeb — CHANGES_REQUESTED** [review]

File: packages/dht/src/dht/DhtNode.ts [hunk 3]
```diff
@@ -143,10 +143,22 @@ export class DhtNode extends EventEmitter<Events> implements ITransport {
...
         this.send = this.send.bind(this)
     }
 
+    private validateConfig(): void {
+        const expectedNodeIdLength = KADEMLIA_ID_LENGTH_IN_BYTES * 2
+        if (this.config.nodeId !== undefined ) {
+            if (!/^[0-9a-fA-F]+$/.test(this.config.nodeId)) {
+                throw new Error('Invalid nodeId, the nodeId should be a hex string')
+            } else if (this.config.nodeId.length !== expectedNodeIdLength) {
+                throw new Error(`Invalid nodeId, the lenght of the nodeId should be ${expectedNodeIdLength}`)
```
Comment (harbu) [inline packages/dht/src/dht/DhtNode.ts]:
```suggestion
                throw new Error(`Invalid nodeId, the length of the nodeId should be ${expectedNodeIdLength}`)
```

**Review by harbu — APPROVED** [review]

**Review by teogeb — APPROVED** [review]

**Review by ptesavol — APPROVED** [review]
Looing good

**Pull request merged**